### PR TITLE
added api endpoints for CRUD

### PR DESCRIPTION
**Pull request by orenpinkas**
- added get posts endpoint
- added create, update and delete post endpoints
- rewrote endpoints with vanilla django approach

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    try:
+        post = Post.objects.get(pk=pk)
+    except Post.DoesNotExist:
+        return HttpResponse(status=404)
+
+    if request.method == "GET":
+        serializer = PostSerializer(post)
+        return JsonResponse(serializer.data)
+
+    elif request.method == "PUT":
```
Comment (oded-regev) [inline api/views.py]: I think we also want to have a POST method here.. why is it missing?

**Review by oded-regev — COMMENTED** [review]
Hi Oren, all in all it looks good, nice work :) 

Please implement CRUD for "author" and "category" as well.

File: api/views.py (hunk 1 above)
Comment (orenpinkas) [reply api/views.py]: I was thinking that the endpoint for POSTing shouldn't be related to any particular post, but I guess I can have it set up here as well for convenience

Comment (oded-regev) [reply api/views.py]: yes it's an expected practice to have an ability to create, edit, delete or get in the same endpoint, that's exactly what CRUD means :)

**Pull request merged**